docs/architecture/adr/ADR037-per-namespace-gateways.md

```diff
@@ -0,0 +1,28 @@
+# ADR037: Per namespace istio gateways
+
+## Status
+
+Pending
+
+## Context
+
+Different tenant applications have differing ingress and egress requirements. Examples include:
+
+* mutual TLS
+* egress controls (e.g. limiting what can egress from a namespace / cluster)
+* certificate operations (renewal etc.)
+
+At present, a single ingress gateway and egress gateway are defined by the GSP. In the case of the ingress gateway, all ingress traffic is routed through this which creates a range of difficulties:
+
+* secrets relating to TLS certs (mutual or otherwise) have to be stored in `istio-system`
+* having some endpoints use mutual TLS and others using non-mutual TLS is difficult/awkward/impossible on a single gateway
```

Comment:
can you expand on the difficulties? I thought there wasn't any problem with having both mTLS and regular TLS on the same ingressgateway.